server/controllers/finance/purchase.js

```diff
@@ -3,152 +3,226 @@ var db = require('./../../lib/db');
 
-// GET /purchaseorders
 
-/*
- * HTTP Controllers
-*/
-exports.execute = function (req, res, next) {
-  executePurchase(req.session.user.id, req.body)
-  .then(function (id) {
-    res.status(200).send({ purchaseId : id });
-  })
-  .catch(next)
-  .done();
-};
 
-exports.getPurchaseOrders = function (req, res, next) {
-  'use strict';
+/** 
+ * Utility method to ensure purchase items lines reference purchase.
+ * @param {Object} purchaseItems - An Array of all purchase items to be written 
+ * @param {string} purchaseUuid - UUID of referenced purchase order
+ * @returns {Object} An Array of all purchases items with guaranteed UUIDs and Purchase orders references
+ */
+function linkPurchaseItems(purchaseItems, purchaseUuid) { 
+  return purchaseItems.map(function (purchaseItem) { 
+     
+    purchaseItem.uuid = purchaseItem.uuid || uuid();
+    purchaseItem.purchase_uuid = purchaseUuid;
+   
+    // Collapse sale item into Array to be inserted into database
+    return Object.keys(purchaseItem).map(function (key) { 
+      return purchaseItem[key];
+    });
+  });
+}
 
-  var sql,
-      status = req.query.status;
-
-  // base query
-  sql = 'SELECT COUNT(uuid) AS count FROM purchase WHERE is_donation = 0 ' +
-        'AND (is_return = 0 OR is_return IS NULL) ';
-
-  switch (status) {
-
-    // the purchase order is pending payment
-    case 'pending':
-      sql += ' AND paid = 0 AND confirmed = 0;';
-      break;
-
-    // the purchase order has been paid, but items have not been
-    // shipped
-    case 'paid':
-      sql += ' AND paid = 1;';
-      break;
-
-    // the purchased items have been shipped, but have not
-    // arrived in the warehouse yet
-    case 'shipped':
-      sql += ' AND closed = 0 AND confirmed = 1;';
-      break;
-
-    // the purchase order has been fulfilled with stock in the
-    // warehouse.
-    case 'delivered':
-      sql += ' AND closed = 1 AND confirmed = 1;';
-      break;
-
-    default:
-      sql += ';';
-      break;
-  }
+// looks up a single purchase record and associated purchase_items
+function lookupPurchaseOrder(uuid, codes) {
+  'use strict';
 
-  db.exec(sql)
+  var record;
+
+  var sql =
+    'SELECT purchase.uuid, purchase.reference, purchase.cost, purchase.discount, purchase.purchase_date, purchase.paid, ' +
+    'creditor.text, employee.name, employee.prenom, user.first, user.last, ' +
+    'purchase.creditor_uuid, purchase.timestamp, purchase.note, purchase.paid_uuid, purchase.confirmed, purchase.closed, ' +
+    'purchase.is_direct, purchase.is_donation, purchase.emitter_id, purchase.is_authorized, purchase.is_validate, ' +
+    'purchase.confirmed_by, purchase.is_integration, purchase.purchaser_id, purchase.receiver_id ' +
+    'FROM purchase ' +
+    'JOIN creditor ON creditor.uuid = purchase.creditor_uuid ' +
+    'JOIN employee ON employee.id = purchase.purchaser_id ' +
+    'JOIN user ON user.id = purchase.emitter_id ' +
+    'WHERE purchase.uuid = ? ;';
+
+  var sqlPurchase =
```

Comment:
Why do you have `sql` and `sqlPurchase`? They look almost identical!  Can't these be the same query?